lychee-lib/src/client.rs

```diff
@@ -654,14 +661,32 @@ impl Client {
         let Ok(path) = uri.url.to_file_path() else {
             return ErrorKind::InvalidFilePath(uri.clone()).into();
         };
-        if !path.exists() {
-            return ErrorKind::InvalidFilePath(uri.clone()).into();
-        }
-        if self.include_fragments {
-            self.check_fragment(&path, uri).await
+        if path.exists() {
+            if self.include_fragments {
+                return self.check_fragment(&path, uri).await;
+            }
         } else {
-            Status::Ok(StatusCode::OK)
+            if self.check_extensions.is_empty() || path.extension().is_some() {
```

Comment:
remove the check `self.check_extensions.is_empty()` it is not necessary since the below logic handles the case already

Reply:
Done.